contracts/NFTVault.sol

```diff
@@ -174,53 +190,69 @@ contract NFTVault is IERC721Receiver, IERC1155Receiver, AccessControl {
      * Token ID and contract address of ERC721 token are saved for receiveToken and backDoor
      */
     function onERC721Received(
-        address _operator,
+        address,
         address _from,
         uint256 _tokenId,
         bytes memory
     ) public virtual override returns (bytes4) {
-        if (keys.transfer(_from, TOKEN_AMOUNT) && whiteList[msg.sender]) {
-            nfts.push(NFT("ERC721", msg.sender, _tokenId, 0, 0));
-            return this.onERC721Received.selector;
+        if (whiteList[_from]) {
+            if (keys.transfer(_from, TOKEN_AMOUNT)) {
+                nfts.push(NFT(msg.sender, _tokenId, 0));
+                return this.onERC721Received.selector;
+            }
         }
     }
 
     /**
      * @dev Emitted when ERC1155 token is received (must be included for receiving ERC1155 tokens)
      *
      * Requirements:
-     * _operator: ERC721 contract address
      * _from: Wallet that sent the NFT
-     * _tokenId:
-     * memory:
+     * _id: ERC1155 token id
+     * _value: ERC1155 value
+     * data:
      *
      * If the ERC1155 contract is whitelisted and keys are sent to _from:
      * Token ID and contract address of ERC721 token are saved for receiveToken and backDoor
      */
     function onERC1155Received(
-        address _operator,
+        address,
         address _from,
         uint256 _id,
         uint256 _value,
         bytes calldata _data
     ) public virtual override returns (bytes4) {
-        if (keys.transfer(_from, TOKEN_AMOUNT) && whiteList[msg.sender]) {
-            nfts.push(NFT("ERC1155", msg.sender, 0, _id, _value));
-            return this.onERC1155Received.selector;
+        if (whiteList[_from]) {
+            if (keys.transfer(_from, TOKEN_AMOUNT)) {
+                nfts.push(NFT(msg.sender, _id, _value));
+                return this.onERC1155Received.selector;
+            }
         }
     }
 
+    /**
+     * @dev Emitted when multiple ERC1155 tokens are received (must be included for receiving ERC1155 tokens)
+     *
+     * Requirements:
+     * _from: Wallet that sent the NFT
+     * _ids: ERC1155 token ids
+     * _values: ERC1155 values
+     * memory:
+     *
+     * If the ERC1155 contract is whitelisted and keys are sent to _from:
+     * Token ID and contract address of ERC721 token are saved for receiveToken and backDoor
+     */
     function onERC1155BatchReceived(
-        address _operator,
+        address,
```

Comment:
- Params naming style differs from rest of the params in the contract (abolish leading `_`).  

- Do not let unnamed param in the function declaration. 

- When checked [token/ERC1155/IERC1155Receiver.sol](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/IERC1155Receiver.sol):

claims operator == msg.sender, probably would be good to have additional test in test/KEYToken.js to be sure you really store contract address of recievied NFT as desired